docs/api-reference/next/server.md

```diff
@@ -148,6 +138,55 @@ export function middleware(req: NextRequest) {
 }
 ```
 
+## userAgent
+
+The `userAgent` helper allows you to interact with the user agent object from the request. It is abstracted from the native `Request` object, and is an opt in feature. It has the following properties:
+
+```typescript
+interface UserAgent {
+  isBot: boolean
+  ua: string
+  browser: {
+    name?: string
+    version?: string
+  }
+  device: {
+    model?: string
+    type?: string
+    vendor?: string
+  }
+  engine: {
+    name?: string
+    version?: string
+  }
+  os: {
+    name?: string
+    version?: string
+  }
+  cpu: {
+    architecture?: string
+  }
+}
+```
```

Comment:
TypeScript is great, but not everyone is writing TypeScript. IMHO we should not use TypeScript to replace the description of each property.

```suggestion
- `isBot`: Whether the request comes from a known bot.
- `browser`
  - `name`: The name of the browser, could be `undefined`.
  - version: The version of the browser, determined dynamically, could be `undefined`.
- `device`
  - `model`: The model of the device, determined dynamically, could be `undefined`.
  - `type`: The type of the browser, can be one of the following values: `console`, `mobile`, `tablet`, `smarttv`, `wearable`, `embedded`, could be `undefined`.
  - `vendor`: The vendor of the device, determined dynamically, could be `undefined`.
- engine
  - name: The name of the browser engine, could be one of the following values: `Amaya`, `Blink`, `EdgeHTML`, `Flow`, `Gecko`, `Goanna`, `iCab`, `KHTML`, `Links`, `Lynx`, `NetFront`, `NetSurf`, `Presto`, `Tasman`, `Trident`, `w3m`, `WebKit`. Could also be `undefined`.
  - version: The version of the browser engine, determined dynamically, could be `undefined`.
- os
  - name: The name of the OS, could be `undefined`.
  - version: The version of the OS, determined dynamically, could be `undefined`.
- cpu
  - architecture: The architecture of the CPU, could be one of the following values: `68k`, `amd64`, `arm`, `arm64`, `armhf`, `avr`, `ia32`, `ia64`, `irix`, `irix64`, `mips`, `mips64`, `pa-risc`, `ppc`, `sparc`, `sparc64`. Could also be `undefined`.
```